src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -251,106 +366,198 @@ bool ShouldAnalyze(SymbolStartAnalysisContext context, INamedTypeSymbol namedTyp
+                    return false;
+
+                var initializer = @this.GetFieldInitializer(variableDeclarator, cancellationToken);
+                if (initializer != null && !@this.SupportsPropertyInitializer(compilation))
+                    return false;
+
+                if (!@this.CanConvert(property))
+                    return false;
+
+                // Can't remove the field if it has attributes on it.
+                var attributes = getterField.GetAttributes();
+                if (attributes.Length > 0 && !@this.SupportsFieldAttributesOnProperties)
+                    return false;
+
+                return true;
+            },
+            (this, compilation, containingType, property, cancellationToken));
+
+        if (getterFields.IsEmpty)
             return;
 
+        var isTrivialSetAccessor = false;
+
         // A setter is optional though.
-        var setMethod = property.SetMethod;
-        if (setMethod != null)
+        if (property.SetMethod != null)
         {
-            var setterField = GetSetterField(semanticModel, setMethod, cancellationToken);
-            // If there is a getter and a setter, they both need to agree on which field they are 
-            // writing to.
-            if (setterField != getterField)
+            // Figure out all the fields written to in the setter.
+            var setterFields = GetSetterFields(semanticModel, property.SetMethod, fieldNames, cancellationToken);
+
+            // Intersect these to determine which fields both the getter and setter write to.
+            getterFields = getterFields.Where(
+                static (field, setterFields) => setterFields.Contains(field),
+                setterFields);
+
+            // If there is a getter and a setter, they both need to agree on which field they are writing to.
+            if (getterFields.IsEmpty)
                 return;
-        }
 
-        var initializer = GetFieldInitializer(variableDeclarator, cancellationToken);
-        if (initializer != null && !SupportsPropertyInitializer(compilation))
-            return;
+            isTrivialSetAccessor = setterFields.TrivialField != null;
+        }
 
-        // Can't remove the field if it has attributes on it.
-        var attributes = getterField.GetAttributes();
-        var suppressMessageAttributeType = compilation.SuppressMessageAttributeType();
-        foreach (var attribute in attributes)
+        if (getterFields.Count > 1)
         {
-            if (attribute.AttributeClass != suppressMessageAttributeType)
-                return;
+            // Multiple fields we could convert here.  Check if any of the fields end with the property name.  If
+            // so, it's likely that that's the field to use.
+            getterFields = getterFields.Where(
+                static (field, property) => field.Name.EndsWith(property.Name, StringComparison.OrdinalIgnoreCase),
+                property);
```

Comment:
special heuristic for the very common case where a field/property have virtually the same name (just differing in casing, or having a `_` prefix, etc.).